packages/@aws-cdk/aws-iotevents-actions/lib/set-timer-action.ts

```diff
@@ -0,0 +1,77 @@
+import * as iotevents from '@aws-cdk/aws-iotevents';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+
+/**
+ * Configuration properties of an action to set a timer.
+ */
+export interface SetTimerActionProps {
+  /**
+   * The duration of the timer, in seconds. One of `duration` or `durationExpression` is required.
+   *
+   * The range of the duration is 60-31622400 seconds.
+   * The evaluated result of the duration expression is rounded down to the nearest whole number.
+   * For example, if you set the timer to 60.99 seconds, the evaluated result of the duration expression is 60 seconds.
+   *
+   * @default - none, required if no `durationExpression` is defined.
+   */
+  readonly duration?: cdk.Duration;
+
+  /**
+   * The duration of the timer, in seconds. One of `duration` or `durationExpression` is required.
+   *
+   * You can use a string expression that includes numbers, variables ($variable.<variable-name>),
+   * and input values ($input.<input-name>.<path-to-datum>) as the duration.
+   *
+   * The range of the duration is 60-31622400 seconds.
+   * The evaluated result of the duration expression is rounded down to the nearest whole number.
+   * For example, if you set the timer to 60.99 seconds, the evaluated result of the duration expression is 60 seconds.
+   *
+   * @default - none, required if no `duration` is defined.
+   */
+  readonly durationExpression?: iotevents.Expression;
+}
+
+/**
+ * The action to create a timer with duration in seconds.
+ */
+export class SetTimerAction implements iotevents.IAction {
+  private readonly durationExpression: string | undefined;
+
+  /**
+   * @param timerName the name of the timer
+   * @param props the properties to set duration
+   */
+  constructor(private readonly timerName: string, props: SetTimerActionProps) {
+    if (!props.duration && !props.durationExpression) {
+      throw new Error('Either duration or durationExpression must be specified');
+    }
+    if (props.duration && props.durationExpression) {
+      throw new Error('duration and durationExpression cannot be specified at the same time');
+    }
+    if (props.duration) {
+      const seconds = props.duration.toSeconds();
+      if (seconds < 60) {
+        throw new Error(`duration cannot be less than 60 seconds, got: ${props.duration.toString()}`);
+      }
+      if (seconds > 31622400) {
+        throw new Error(`duration cannot be greater than 31622400 seconds, got: ${props.duration.toString()}`);
+      }
+      this.durationExpression = seconds.toString();
+    }
+    if (props.durationExpression) {
+      this.durationExpression = props.durationExpression.evaluate();
+    }
+  }
+
+  bind(_scope: Construct, _options: iotevents.ActionBindOptions): iotevents.ActionConfig {
```

Comment:
```suggestion
  public _bind(_scope: Construct, _options: iotevents.ActionBindOptions): iotevents.ActionConfig {
```

Reply:
This would be better off as it is I think.
This is implementation of `IAction` and is following [`Integration` in DESIGN_GUIDELINE](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#integrations) as same as [aws-events target](https://github.com/aws/aws-cdk/blob/1af03c9dc3deb0ab01f4f8713b44cd909aaab8cd/packages/@aws-cdk/aws-events/lib/target.ts#L10-L19).

Reply:
Yeah, I'm annoyed with that but it's not in scope here to ask you to fix an old contract I'm mad at. Maybe I'll get neurotic about it and do some excessive refactoring.

Reply:
I've understood that refactoring is necessary even in light of the current implementations.
I'll refactor it!